examples/AvroSpecific/Program.cs

```diff
@@ -0,0 +1,124 @@
+                Console.WriteLine("Usage: .. bootstrapServers schemaregistryurl topicName");
+                return;
+            }
+
+            string bootstrapServers = args[0];
+            string schemaRegistryUrl = args[1];
+            string topicName = args[2];
+
+            var producerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", bootstrapServers },
+                // Note: you can specify more than one schema registry url using the
+                // schema.registry.url property for redundancy (comma separated list). 
+                // The property name is not plural to follow the convention set by
+                // the Java implementation.
+                { "schema.registry.url", schemaRegistryUrl },
+                // optional avro / schema registry client properties:
+                { "avro.buffer.bytes", 50 },
+                { "avro.auto.register.schema", true },
+                { "schema.registry.timeout.ms", 5000 },
+                { "schema.registry.cache.capacity", 10 }
+            };
+            var consumerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", bootstrapServers },
+                { "group.id", Guid.NewGuid() },
+                { "schema.registry.url", schemaRegistryUrl }
+            };
+
+            // Note: Each AvroSerializer and AvroDeserializer instance below internally creates and manages the 
+            // lifecycle of a CachedSchemaRegistry instance, taking it's configuration properties from the 
+            // dictionary supplied to the Producer/Consumer constructor. This is the most straightforward way 
+            // to use these (de)serializers, however you can also pass a CachedSchemaRegistry instance into the 
+            // constructors directly, which avoids duplicate connections to Schema Registry in the case where 
+            // you're using more than one AvroSerializer/Deserializer. E.g.:
+            // 
+            // var schemaRegistryConfig = new Dictionary<string, object>
+            // {
+            //    { "schema.registry.url", schemaRegistryUrl },
+            //    { "avro.buffer.bytes", 50 }
+            // };
+            // 
+            // using (var schemaRegistryClient = new CachedSchemaRegistryClient(schemaRegistryConfig))
+            // using (var consumer = new Consumer<User, User>(consumerConfig, new AvroDeserializer<User>(schemaRegistryClient), new AvroDeserializer<User>(schemaRegistryClient)))
+            // using (var producer = new Producer<User, User>(producerConfig, new AvroSerializer<User>(schemaRegistryClient), new AvroSerializer<User>(schemaRegistryClient)))
+            // {
+            // ...
+
+            // Note: The User class in this project was generated using the Confluent fork of the avrogen.exe tool 
+            // (avaliable from: https://github.com/confluentinc/avro/tree/confluent-fork) which includes modifications
```

Comment:
I'll make our version of avrogen.exe available via some more convenient means at release time.